codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -70,14 +70,19 @@ let make = (
   ~loadLayer: LoadLayer.t,
   ~shouldUseTui=false,
 ) => {
+  let targetBatchesInBuffer = 3
+  let targetBufferSize = switch Env.targetBufferSize {
+    | Some(size) => size
+    | None => Env.maxProcessBatchSize * (chainManager.chainFetchers->ChainMap.size > targetBatchesInBuffer ? 1 : targetBatchesInBuffer)
+  }
```

Comment:
We don't need multiple batches in the buffer, when there are multiple chains.

Reply:
For better memory, it'd be possible to make it less than Env.maxProcessBatchSize per chain. However, I don't want to do it to keep the single-chain batch optimisation working.

Reply:
Nice, do you want to add these notes in a comment for future implementation?